ostree-upload-server.py

```diff
@@ -123,24 +115,41 @@ def upload(self):
                 if upload.filename == "":
                     return "no filename in upload\n", 400
 
-                # TODO: tempdir should be a member
                 (f, real_name) = tempfile.mkstemp(dir=self._tempdir)
                 os.close(f)
                 upload.save(real_name)
 
-                # TODO: task_list reference should be a member
-                self._webapp_callback(upload.filename, real_name)
+                self._webapp_callback(ReceiveTask(upload.filename, real_name, self._repo))
                 logging.debug("/upload: POST request completed for " + upload.filename)
 
                 return "task added\n"
         else:
             return "only POST method supported\n", 400
 
+    def push(self):
+        """
+        Extract a flatpak bundle from local repository and push to a remote
+        """
+        if not self._check_auth():
+            return self._authenticate()
+
+        logging.debug(request.args)
+        try:
+            ref = request.args['ref']
+            remote = request.args['remote']
+        except KeyError:
+            return "ref and remote arguments required", 400
+        logging.debug("/push: {0} to {1}".format(ref, remote))
+        if not remote in self._push_adapters:
+            return "unknown remote", 400
+        adapter = self._push_adapters[remote]
+        self._webapp_callback(PushTask(ref, self._repo, ref, adapter, self._tempdir))
+        return("/push: {0} to {1}".format(ref, remote))
```

Comment:
Messages sent to the client should be a bit more descriptive and/or json format with fields.
i.e. (not sure of syntax here):
```
message = { success = true, message = "Pushed blah package to server_name" }
return (message)
```

and for errors:
```
message = { success = false, error = "Unknown server" message = "Remote server_name is not in the whitelist" }
return (message)
```